### ISSUE TITLE
Json parser throws ArgumentOutOfRangeException when node is repeated

### ISSUE THREAD

**Issue by pranavkm** [aspnet/dnx#1878]
project.json:
```JSON
{
    "foo": {}, "foo": {}
}
```

```
Restoring packages for project.json
----------
Microsoft.Framework.Runtime.FileFormatException: Index (zero based) must be greater than or equal to zero and less than the size of the argument list. ---> System.FormatException: Index (zero based) must be greater than or equal to zero and less than the size of the argument list.
   at System.Text.StringBuilder.AppendFormatHelper(IFormatProvider provider, String format, ParamsArray args)
   at System.String.FormatHelper(IFormatProvider provider, String format, ParamsArray args)
   at System.String.Format(String format, Object[] args)
   at Microsoft.Framework.Runtime.Json.JsonDeserializer.DeserializeObject(JsonToken head, JsonBuffer buffer)
   at Microsoft.Framework.Runtime.Json.JsonDeserializer.DeserializeInternal(JsonToken next, JsonBuffer buffer)
   at Microsoft.Framework.Runtime.Json.JsonDeserializer.Deserialize(TextReader reader)
   at Microsoft.Framework.Runtime.Project.GetProjectFromStream(Stream stream, String projectName, String projectPath, ICollection`1 diagnostics)
   at Microsoft.Framework.Runtime.Project.TryGetProject(String path, Project& project, ICollection`1 diagnostics)
   --- End of inner exception stack trace ---
   at Microsoft.Framework.Runtime.Project.TryGetProject(String path, Project& project, ICollection`1 diagnostics)
   at Microsoft.Framework.PackageManager.RestoreCommand.<RestoreForProject>d__73.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Framework.PackageManager.RestoreCommand.<>c__DisplayClass72_0.<<ExecuteCommand>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Framework.PackageManager.RestoreCommand.<ExecuteCommand>d__72.MoveNext()
----------
Restore failed
Index (zero based) must be greater than or equal to zero and less than the size of the argument list.
```

**Comment by pranavkm** [comment]
I guess one of the format string isn't getting all the right parameters.

**Comment by troydai** [comment]
Exception should be thrown because duplicate member name is not allowed. But the error message looks wrong.

**Comment by pranavkm** [comment]
Yeah, it should throw, but it looks like you aren't passing in all the right format params to a format string.

**Comment by troydai** [comment]
Yes. Found this issue: will be fixed in https://github.com/aspnet/dnx/pull/1872. 
I'll merge that PR soon.

The issue roots here: https://github.com/aspnet/dnx/blob/dev/src/Microsoft.Framework.Runtime.Hosting/Json/JsonDeserializerResource.cs#L48

**Comment by troydai** [comment]
Fixed in 6b0dc2bfeb79a6490f7613c8086c2d6de13a95a6
